frontend-service/src/components/MapBoxMap.tsx

```diff
@@ -1,14 +1,57 @@
 import React, { useEffect, useRef } from 'react';
 import mapboxgl, { MapMouseEvent } from 'mapbox-gl';
 import 'mapbox-gl/dist/mapbox-gl.css';
-import type { MapboxMapProps } from './MapBoxMap.types';
-import { calculateKm, emojiForDistances } from '../utils/mapUtils';
-import { cursorSetup, recentreAndOrZoom } from '../utils/mapboxUtils';
+import type { MapboxMapProps } from './types/MapBoxMap.types';
+import { calculateKm } from '../utils/mapUtils';
+import { cursorSetup, recentreAndOrZoom, addLineToMap, addLineSourceToMap, createDistanceMarkerElement } from '../utils/mapboxUtils';
+import { mapBoxMapStyle } from '../objects/mapBoxConsts';
+import { Pin } from '../components/types/Game.types'
 mapboxgl.accessToken = process.env.GATSBY_MAPBOX_ACCESS_TOKEN as string;
 
 const MapboxMap = ({ roundDetails, handleGuess }: MapboxMapProps) => {
-	const mapContainerRef = useRef(null);
-	const addMarker = (map: mapboxgl.Map, e: MapMouseEvent) => {
+	const mapContainerRef = useRef(null)
+	const mapRef = useRef<mapboxgl.Map | null>(null)
+	const currentLineIdRef = useRef<string>(''); 
```

Comment:
`currentLineIdRef` is the name of the line we are printing to the map when we make a guess. When I was reloading the entire map between guesses the line simply disappeared. Now that I'm not reloading the entire map I need to store the line element as a ref so I know exactly what to delete when the next round loads.